proposals/0035-linalg-matrix.md

```diff
@@ -513,6 +547,32 @@ expected target data format. When read from `groupshared` memory, the data may
 be in any arithmetic or packed data type. If the type mismatches the target data
 type of the matrix a data conversion is applied on load.
 
+#### Matrix::FromThreadVectors
+
+```c++
+template <MatrixUse UseLocal = Use>
+typename hlsl::enable_if<Use == MatrixUse::A && Scope == MatrixScope::Wave &&
+                              UseLocal == Use,
+                          Matrix>::type
+    FromThreadVectors(vector<ElementType, MScalars>);
+
+template <MatrixUse UseLocal = Use>
+typename hlsl::enable_if<Use == MatrixUse::B && Scope == MatrixScope::Wave &&
+                              UseLocal == Use,
+                          Matrix>::type
+    FromThreadVectors(vector<ElementType, NScalars>);
+```
+
+Produces a matrix from per-thread vectors. An A matrix is produced from
+per-thread row vectors, while a B matrix is produced from per-thread column
+vectors. The `FromThreadVectors` construction method is not available for
+accumulator matrices which vary by hardware implementation.
+
+When creating an A matrix, the N dimension must be less than or equal to the
+wave size. When creating a B matrix, the M dimension must be less than or equal
+to the wave size. Threads outside the matrix size are discarded.
+
+Must be called from wave-uniform control flow.

```

Comment:
This seems like an unnecessary restriction. Wouldn't it be possible to partially pull a matrix from masked threads? Not 100% sure the use case (identity matrix loads, maybe?) but enforcing this might be power intensive if you only need certain portions of a matrix.

Reply:
I think we'd need to define what values get pulled from masked threads. Is it 0's or... something else?